suewsmodel/suews_wrapper.py

```diff
@@ -50,6 +50,29 @@ def wrapper(pathtoplugin, year=None):
     # SuPy simulation
     df_output, df_state_final = sp.run_supy(df_forcing, df_state_init, check_input=True, serial_mode=True)
 
+    # save user input and output as diagnostics
+    try:
+        import pandas as pd
+
+        p_dir_dump = Path(pathtoplugin) / "diagnostics"
+        if not p_dir_dump.exists():
+            p_dir_dump.mkdir(parents=True)
+
+        str_now = pd.Timestamp.now().isoformat().split('.')[0]
+        ver_supy=sp.__version__
+        fn_dump = p_dir_dump/f"supy{ver_supy}_diag_{str_now}.h5"
+        df_forcing.to_hdf(fn_dump, key="df_forcing")
+        df_output.to_hdf(fn_dump, key="df_output")
+        df_state_final.to_hdf(fn_dump, key="df_state_final")
+        # tell user the location where `fn_dump` is saved
+        print(f"{fn_dump.resolve()} has been successfully created!")

```

Comment:
this line may be removed: apparently a leftover when I was testing it under a pure python environment.